add_node.rb

```diff
@@ -10,8 +11,53 @@
 fileData = File.read("data.json")
 obj = JSON.parse(fileData)
 
+# Initialise time periods.Dates have the format of yyyy-mm-dd
+this_year = Date.today.strftime("%Y")
+MID_SPRING_PERIOD = ["#{this_year}-02-15","#{this_year}-04-30"]
+END_SPRING_PERIOD = ["#{this_year}-05-01","#{this_year}-09-30"]
+MID_AUTUMN_PERIOD = ["#{this_year}-10-01","#{this_year}-11-30"]
+END_AUTUMN_PERIOD = ["#{this_year}-12-01","#{(this_year.to_i+1).to_s}-02-14"]
+
 puts "#{obj.length} objects earlier!"
 
+today = Date.today.to_s
+# Find out default semester & year
+if today >= MID_SPRING_PERIOD[0] && today <= MID_SPRING_PERIOD[1]
+	default_semester = "mid spring "+this_year
+elsif today >= END_SPRING_PERIOD[0] && today <= END_SPRING_PERIOD[1]
+	default_semester = "end spring "+this_year
+elsif today >= MID_AUTUMN_PERIOD[0] && today <= MID_AUTUMN_PERIOD[1]
+	default_semester = "mid autumn "+this_year
+elsif today >= END_AUTUMN_PERIOD[0]	&& today <= END_AUTUMN_PERIOD[1]
+	default_semester = "end autumn "+this_year
+else
+	puts "Time period initialisation seems wrong."
+	break	 
+end
+
+
+# Just make sure that batch is not "" as while loop will never execute
+batch = "h"
+batch_semester = default_semester
+
+while !["y","Y","n","N"].include? batch
+	puts "Do you want to batch insert year & semester for the papers? [y/Y for Yes, n/N for No) "
+	batch = gets.chomp
+	if batch.downcase == "y"
+		puts "Enter batch semester (#{default_semester}) : "
+		batch_semester = gets.chomp
+		if batch_semester.length == 0 
+			batch_semester = default_semester
```

Comment:
Before doing this, make sure that the semester entered by the user matches [this](https://github.com/metakgp/mfqp-source/blob/master/scripts/upload/upload.rb#L18) regex. If not, keep them in the loop, because they probably want batch entry but just entered it wrong. (Remove the ending `pdf` part from the regex on that script)